### PR TITLE
API SS4 compatibility

### LINKED ISSUE #14: Implement namespaces and update for SS4

### LINKED ISSUE #15: Update documentation
Ensure new examples are correct in SS4 context

### PR DESCRIPTION
**Pull request by robbieaverill**
High level changes:

* Implement SS4 namespaces and update composer requirements
* Change some of the extension hooks to match what is in SS4 now
* Remove MonologSyslogWriter, users should use Monolog handlers instead now

Resolves #14 and resolves #15

### CODE REVIEW

File: code/AuditHook.php [hunk 1]
```diff
@@ -122,12 +146,12 @@ public static function handle_manipulation($manipulation)
             }
 
             // log PermissionRole being added to a Group
-            if ($table == 'Group_Roles') {
-                $role = \PermissionRole::get()->byId($details['fields']['PermissionRoleID']);
-                $group = \Group::get()->byId($details['fields']['GroupID']);
+            if ($table === 'Group_Roles') {
```
Comment (NightJar) [inline code/AuditHook.php]: I know it's core so should be good, but would this not be better to be using `DataObjectSchema`? Technically it is a config var, and thus could be altered.

File: code/AuditHook.php [hunk 2]
```diff
@@ -145,12 +169,12 @@ public static function handle_manipulation($manipulation)
             }
 
             // log Member added to a Group
-            if ($table == 'Group_Members') {
-                $member = \Member::get()->byId($details['fields']['MemberID']);
-                $group = \Group::get()->byId($details['fields']['GroupID']);
+            if ($table === 'Group_Members') {
```
Comment (NightJar) [inline code/AuditHook.php]: as per previous `DataObjectSchema` question

File: code/AuditHookManyManyList.php [hunk 3]
```diff
@@ -13,13 +18,13 @@ public function removeByID($itemID)
         parent::removeByID($itemID);
 
         if ($this->getJoinTable() == 'Group_Members') {
-            $currentMember = \Member::currentUser();
+            $currentMember = Member::currentUser();
```
Comment (NightJar) [inline code/AuditHookManyManyList.php]: `Security::getCurrentUser`

File: tests/AuditHookTest/Logger.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace SilverStripe\Auditor\Tests\AuditHookTest;
+
+use Psr\Log\AbstractLogger;
+
+class Logger extends AbstractLogger
```
Comment (NightJar) [inline tests/AuditHookTest/Logger.php]: Stub should be `implements TestOnly`?

**Review by NightJar — COMMENTED** [review]
A few minor things that I should have probably put as a review but for some reason logged individual comments.

File: code/AuditHook.php (hunk 1 above)
Comment (robbieaverill) [reply code/AuditHook.php]: I'm not sure we can, this is a pivot table rather than a physical model

Comment (robbieaverill) [reply code/AuditHook.php]: Parent table plus relation name - we can

**Pull request merged**